src/backend/joanie/tests/testing_utils.py

```diff
@@ -490,6 +513,37 @@ def generate(self):  # pylint: disable=too-many-locals,too-many-statements
             "Successfully create an order for a PRODUCT_CREDENTIAL "
             "with an installment payment failed"
         )
+        # Order for PRODUCT_CERTIFICATE with voucher discount rate of 10%
+        order_certificate = self.create_product_purchased(
+            student_user,
+            organization_owner,
+            organization,
+            enums.PRODUCT_TYPE_CERTIFICATE,
+            enums.ORDER_STATE_COMPLETED,
+        )
+        self.add_discount(order_certificate, discount_rate=0.1)
```

Comment:
I think that the discount should be defined in create_product_purchased
something like
```suggestion
        order_certificate = self.create_product_purchased(
            student_user,
            organization_owner,
            organization,
            enums.PRODUCT_TYPE_CERTIFICATE,
            enums.ORDER_STATE_COMPLETED,
            discount_rate=0.1
        )
```